src/core/statistics/useStatistics.ts

```diff
@@ -0,0 +1,52 @@
+// Useful resources for Mapp tracking:
+// https://documentation.mapp.com/1.0/en/manual-track-request-25105181.html
+// https://documentation.mapp.com/1.0/en/how-to-send-manual-tracking-requests-page-updates-7240681.html
+
+export type CustomClickParameter = {
+  [key: number]: number | { [key: number]: string };
+};
+
+export type EventData = {
+  linkId: string;
+  customClickParameter: CustomClickParameter;
+};
+
+// 1. page - when a page request is simulated
+// 2. click - for measuring actions that don't cause page load;
+// 3. link - clicking a link that triggers a new page load
+// 4. pageupdate - information on the page changes without a new page load
+type EventType = "page" | "click" | "link" | "pageupdate";
+
+export function useStatistics(eventType: EventType) {
+  // If the global wts object doesn't exist, it means we are in dev environment.
+  // Here instead of actually tracking we just log the data to the console.
+  if (!window.wts) {
+    window.wts = {
+      push(trackingProps: ["send", EventType, EventData]) {
+        // eslint-disable-next-line no-console
+        console.log(
+          `Tracking: ${trackingProps[0]}, ${trackingProps[1]}, ${JSON.stringify(
+            trackingProps[2]
+          )}`
+        );
+      }
+    };
+  }
+
+  return {
+    track: (
+      id: number,
+      name: string,
+      trackedData: string | number | string[]
+    ) => {
+      const eventData: EventData = {
+        linkId: name,
+        customClickParameter: {}
+      };
+      eventData.customClickParameter[id] = trackedData;
+      window.wts.push(["send", eventType, eventData]);
```

Comment:
Without specific hooks per event I do not think this signature works well.

As mentioned previous `linkId` and `customClickParameter` are not required across event types.

I do not think it is helpful to specify the event type when invoking the hook either. It might as well be sent when invoking the `track()` method.

I suggest something along the following signature for the track method:

```typescript
export type Parameters = {
  [key: string]: number | string | { [key: number]: string | number };
};

function track(eventType: EventType, parameters: Parameters) {}
```

(Note that this is totally untested)

Reply:
I moved specifying the event type down to the track() function, but specified the parameters more strictly than you describe. With the current setup we assume that we will be needing specifically the id, name, and trackedData, which also needs to be sent to Mapp in a very specific format.